modules/backend/widgets/form/assets/js/october.form.js

```diff
@@ -220,6 +251,157 @@
             .nextUntil('.section-field').hide()
     }
 
+    /*
+     * Bind validation checks on available fields, triggered when
+     * the fields is blurred or changed
+     */
+    FormWidget.prototype.bindValidation = function () {
+        var fields = this.getFieldElementsToValidate(),
+            $form = this.$form
+
+        for (var i in fields) {
+            var field = fields[i]
+
+            if (!field || !field.className) {
+                continue
+            }
+
+            var fieldClasses = field.className.split(/\s+/).map(function (className) {
```

Comment:
@bennothommo I think this part is overly complicated :) 
If I've understood it correctly you need to find the field which has a class corresponded to the Regexp - `/^([a-z\-]+)-field$/` and that's it.

Two options: either use `indexOf()` to cut down all those not needed fields or change Regex to be more accurate (before&after spaces, dirty try: `(\s+)?([a-z\-]+)-field(\s+)?`).

Reply:
@w20k Yeah I did initially have just a regex, but I thought this way was more readable, albeit a bit more complex. Your regex works just as well though. :)